src/napari/layers/base/base.py

```diff
@@ -907,7 +914,8 @@ def affine(self, affine: npt.ArrayLike | Affine) -> None:
         self._transforms[2] = coerce_affine(
             affine, ndim=self.ndim, name='physical2world'
         )
-        self.refresh()
+        self._clear_extent()
+        self.refresh(extent=False)
         self.events.affine()
```

Comment:
I don't understand this change. Before and after are equivalent, as far as I can tell. Same for all the others?

Reply:
If you execute:
```
with layer._block_refresh():
    layer.affine = new_value
```

The code is not equivalent. 
For current main, the `extent is not invalidated when callback is triggered.